transport/internal/tls/muxlistener/tls_checker.go

```diff
@@ -44,8 +44,9 @@ const (
 // record in the initial bytes.
 // Read more about header spec: https://datatracker.ietf.org/doc/html/rfc8446#section-5.1
 func isTLSClientHelloRecord(r io.Reader) (bool, error) {
-	buf := make([]byte, _tlsHandshakeHeaderLength)
-	n, err := r.Read(buf)
+	// Use array to avoid slice allocation
+	var buf [_tlsHandshakeHeaderLength]byte
+	n, err := r.Read(buf[:])
```

Comment:
I remember trying this out earlier, compiler most likely moves it to the heap as it is passed a slice to an interface. At times compiler skips moving it to the heap if the method called does not store `b`, in this case it's not possible as `r` is an interface.

Reply:
Ran compiler analysis quickly on the above code:
```
tls_checker.go:46:29: leaking param: r
tls_checker.go:48:6: moved to heap: buf
```

Reply:
Interesting! Didn't realise interface methods end up defaulting to heap :/